test/cluster/storage_test.go

```diff
@@ -352,6 +354,174 @@ var _ = Describe(
+												},
+											},
+										},
+									},
+								}
+
+								aeroCluster.Spec.Storage.Volumes = append(aeroCluster.Spec.Storage.Volumes, hostPathVolume)
+
+								// Deploy the cluster
+								err := DeployCluster(k8sClient, ctx, aeroCluster)
+								Expect(err).ToNot(HaveOccurred())
+
+								// Test Aerospike server container mount options
+								podNamespacedName := test.GetNamespacedName(aeroCluster.Name+"-0-1", aeroCluster.Namespace)
+
+								// Validate Aerospike server container mount options
+								aerospikeMountOptions, err := getVolumeMountOptions(podNamespacedName,
+									asdbv1.AerospikeServerContainerName, "hostpath-mount-options-test")
+								Expect(err).ToNot(HaveOccurred())
+
+								Expect(asdbv1.GetBool(aerospikeMountOptions.ReadOnly)).To(BeTrue())
+								Expect(aerospikeMountOptions.SubPath).To(Equal("subdir"))
+								Expect(aerospikeMountOptions.SubPathExpr).To(BeEmpty())
+								Expect(*aerospikeMountOptions.MountPropagation).To(Equal(v1.MountPropagationHostToContainer))
+
+								// Validate init container mount options
+								initMountOptions, err := getVolumeMountOptions(podNamespacedName,
+									asdbv1.AerospikeInitContainerName, "hostpath-mount-options-test")
+								Expect(err).ToNot(HaveOccurred())
+
+								Expect(asdbv1.GetBool(initMountOptions.ReadOnly)).To(BeTrue())
+								Expect(initMountOptions.SubPath).To(Equal("subdir"))
+								Expect(aerospikeMountOptions.SubPathExpr).To(BeEmpty())
+								Expect(*initMountOptions.MountPropagation).To(Equal(v1.MountPropagationHostToContainer))
+
+								// Validate custom init container mount options
+								customInitMountOptions, err := getVolumeMountOptions(podNamespacedName,
+									"custom-init-container", "hostpath-mount-options-test")
+								Expect(err).ToNot(HaveOccurred())
+
+								Expect(asdbv1.GetBool(customInitMountOptions.ReadOnly)).To(BeTrue())
+								Expect(customInitMountOptions.SubPath).To(Equal("custom-init-subdir"))
+								Expect(aerospikeMountOptions.SubPathExpr).To(BeEmpty())
+								Expect(*customInitMountOptions.MountPropagation).To(Equal(v1.MountPropagationHostToContainer))
+
+								// Validate sidecar container mount options
+								sidecarMountOptions, err := getVolumeMountOptions(podNamespacedName,
+									"sidecar-container", "hostpath-mount-options-test")
+								Expect(err).ToNot(HaveOccurred())
+
+								Expect(asdbv1.GetBool(sidecarMountOptions.ReadOnly)).To(BeTrue())
+								Expect(sidecarMountOptions.SubPath).To(Equal("sidecar-subdir"))
+								Expect(aerospikeMountOptions.SubPathExpr).To(BeEmpty())
+								Expect(*sidecarMountOptions.MountPropagation).To(Equal(v1.MountPropagationHostToContainer))
+
+								volumeIndex := len(aeroCluster.Spec.Storage.Volumes) - 1
+
+								// Test with read-write mount (should fail due to validation)
+								aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+								Expect(err).ToNot(HaveOccurred())
```

Comment:
We don't need to fetch the cluster again and again. `updateCluster` func already does that internally